#!/usr/bin/env python
PACKAGE = "bitbots_vision_tools"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

group_color_space_finder = gen.add_group("ColorSpaceFinder", type="tab")
group_ROS = gen.add_group("ROS", type="tab")
group_color_detector = gen.add_group("ColorDetector", type="tab")

group_pixel_list_color_detector = group_color_detector.add_group("PixelListColorDetector")
group_hsv_color_detector = group_color_detector.add_group("HsvColorDetector")

group_color_space_finder.add("debug_on",   bool_t,   0, "Turns debug-mode on",  True)

group_ROS.add("ROS_img_msg_topic", str_t, 0, "ROS_img_msg_topic", None)
group_ROS.add("ROS_img_queue_size", int_t, 0, "ROS_img_queue_size", min=1, max=20)

group_color_detector.add("ColorDetector", str_t, 0, "choose which color detector to use", None)

gen.add("h_min",    int_t,    0, "h_min", 0,  0, 255)
gen.add("s_min",    int_t,    0, "s_min", 0,  0, 255)
gen.add("v_min",    int_t,    0, "v_min", 0,  0, 255)
gen.add("h_max",    int_t,    0, "h_max", 255,  0, 255)
gen.add("s_max",    int_t,    0, "s_max", 255,  0, 255)
gen.add("v_max",    int_t,    0, "v_max", 255,  0, 255)

group_pixel_list_color_detector.add("pixel_list_color_detector_path",    str_t,    0, "pixel_list_color_detector_path", None)

exit(gen.generate(PACKAGE, "color_space_finder", "ColorSpaceFinder"))
